proposals/12020-model-registry-integration/README.md

```diff
@@ -0,0 +1,432 @@
+enhancement.
+
+#### Option 1: Extend the kfp-launcher ConfigMap
+
+Extend the existing `kfp-launcher` ConfigMap to include Model Registry configuration:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kfp-launcher
+  namespace: kubeflow
+data:
+  # Existing configuration...
+  pipelineRoot: "s3://kfp-artifacts"
+
+  # New Model Registry configuration
+  modelRegistry: |
+    url: https://model-registry.example.com:8443
+    tokenSecretRef: # Follows the same key names for existing configurations
+      secretName: model-registry-auth
+      secretNamespace: model-registry-system  # Defaults to current namespace
+      tokenKey: token
+    caConfigMapRef:  # Optional TLS certificate
+      configMapName: model-registry-ca-bundle
+      configMapNamespace: model-registry-system
+      key: ca-bundle.crt
+    timeout: 30s  # HTTP timeout for registration requests
+    retryAttempts: 3  # Number of retry attempts on failure
+```
+
+#### Option 2: Introduce LauncherConfig CRD
+
+Create a dedicated Custom Resource Definition for launcher configuration:
+
+```yaml
+apiVersion: pipelines.kubeflow.org/v2beta1
+kind: LauncherConfig
+metadata:
+  name: kfp-launcher
+spec:
+  modelRegistry:
+    url: https://model-registry.example.com:8443
+    tokenSecretRef:
+      name: model-registry-auth
+      namespace: model-registry-config # defaults to the current namespace
+      key: token
+    caConfigMapRef: # optional
+      name: model-registry-ca-bundle
+      namespace: model-registry-namespace # defaults to the current namespace
+      key: ca-bundle.crt
+    timeout: 30s
+    retryAttempts: 3
+```
+
+### User Stories
+
+#### Story 1
+
+As a data scientist, I would like to register my fine-tuned model to Model Registry in a pipeline without knowing the
+underlying Model Registry infrastructure and APIs, so that I can easily track model versions, share models with my team,
+and maintain a centralized model catalog as part of my automated ML workflows.
+
+#### Story 2
+
+As a data scientist, I would like to access a centralized model catalog that is independent of the specific tool used to
+create the models, enabling easy discovery and simplified model management across different ML workflows.
+
+### Notes/Constraints/Caveats
+
+N/A
+
+### Risks and Mitigations
+
+1. The Model Registry API doesn't have granular RBAC so the `pipeline-runner` service account has full access to the
+   Model Registry API.
+
+## Design Details
+
+### Architecture Overview
+
+```
+┌─────────────────┐     ┌──────────────────┐     ┌─────────────────┐
+│   KFP Pipeline  │     │   KFP Launcher   │     │ Model Registry  │
+│                 │     │                  │     │                 │
+│ model.register()│───▶│ Extract metadata │───▶│   API Server    │
+│                 │     │ Register model   │     │                 │
+└─────────────────┘     └──────────────────┘     └─────────────────┘
+                              │
+                              ▼
```

Comment:
I think the arrow should be reversed since we're reading in the configuration and not writing to